### DISCUSSION TITLE
Option to disable setting readback?

### DISCUSSION THREAD

**Discussion by ShaneMurphy2** [serialport/serialport-rs#164] [Ideas]
Some background, I'm working with a Xilinx UARTLite IP. This serial device is very barebones, and doesn't support all features. As such, the driver handles calls to `tcsetattr` a little oddly, in that it will [mutate][1] the settings the user passed in to be what the device supports. This causes an issue with opening this port using this library, since it calls `tcgetattr` after `tcsetattr`. For some combinations of settings, the readback will be different. For this device, that is not a problem. The core, low-level settings are configured in hardware (data bits, flow control, etc.).

The only thing I need this library to do is higher level setup. I recognize I can construct a TTYPort instance from a raw FD, but I was having difficulty getting that to work reliably when constructing a `SerialStream` from `tokio_serial`. 

My proposal is to add a setting to the builder called `best_effort`, that when passed `true` will disable the call to `tcgetattr` and the setting check. So, my question is: is this use case of a devious device driver is common enough to warrant an upstream change like this?

[1]: https://github.com/Xilinx/linux-xlnx/blob/master/drivers/tty/serial/uartlite.c#L324-L327

**Comment by sirhcel** [comment]
Hello @ShaneMurphy2, you are not alone experiencing such behavior. I've seen this for `CRTSCTS` the flag in https://github.com/Fazecast/jSerialComm/issues/453#issuecomment-1407004898. Apparently few people are actually hitting such issues.

What is the actual show stopper for you in your case? Getting an error from [`SerialPortBuilder::open`](https://github.com/serialport/serialport-rs/blob/c3209197dfbc0c760c843c7905d49b83b99714e3/src/posix/tty.rs#L148)?

Interestingly, reading back termios flags only seems to happen `SerialPortBuilder::open` and not in other functions changing them.

**Reply by ShaneMurphy2** [reply]
Yeah, it was opening the port. I got around it by writing my own version of open that ~~steals~~ borrows from `serialport` and uses a fork of [tokio-serial](https://github.com/ShaneMurphy2/tokio-serial/tree/from-tty) that allows constructing a `SerialStream` from a raw `TTYPort`. The version I wrote takes the same settings as `SerialPortBuilder`, but uses them as a sanity check, but does not try to apply them since my particular port is not configurable.
